### ISSUE TITLE
[Feature] @Column() naming convention should be improved

### ISSUE THREAD

**Issue by luis901101** [conduit-dart/conduit#67]
### **Is your feature request related to a problem? Please describe.**

Currently the naming conventions used when creating a column from a model property after `conduit db upgrade` is not camelCase nor snake_case it's in fact the property's name toLowerCase(). Lets say we have this model:

```dart
@Table(name: 'venue')
class _Venue {
  @primaryKey int id;
  String name;
  @Column(nullable: true) String? shortDescription;
}
```

Here the **shortDescription** is a two words property so for better reading and understanding the expected column name should be camelCase -> `shortDescription` or snake_case -> `short_description`, but instead the column name ends up being the two words in lower case with no separation.

### **Describe the solution you'd like**

*In **Postgres documentation** the naming convention used is **snake_case** so maybe this should be the default one.* 

Anyway in order to be impartial on this matter, maybe the best way for naming columns after a model property should be using the same name used in the property... so if developer uses **camelCase** on his model properties then columns should be named using **camelCase** if developer uses **snake_case** for properties the same should be used for columns... just mirroring the property name to the column name.

In addition the developer should be able to name table columns with a naming convention different than model properties.
For instance using the same sample from above:

```dart
@Table(name: 'venue')
class _Venue {
  @primaryKey int id;
  String name;
  @Column(name: 'short_description', nullable: true) String? shortDescription;
}
```

The same way `@Table()` allows to set a custom name `@Column()` should allow the same.

**Comment by Reductions** [comment]
I would like to make the `snake_case` the default one. But the problem with that is we will need one of:
1) A tool either to fix the code
2) A tool that creates a migration that will change the names to the new convention.
3) Make a 4.0 release that brakes compatibility.

**Comment by luis901101** [comment]
I think this will be a necessary breaking change, but maybe can be made with some deprecation concept in mind, I mean, using default `snake_case` for every table and column but also offering some deprecated property like: `legacyNaming: true` in `@Table` to allow using current naming convention for a table and its columns, not breaking code…

Another approach is offering the `name: 'myPropertyName'` I proposed above to let the developer to custom name the properies.

I also think that creating tools for fixing the current code or migrating the naming convention will not be easy at all, it is possible that these tools fails in large scale projects with a complicated logic.

**Comment by luis901101** [comment]
Hi @Reductions, I made this [PR 70](https://github.com/conduit-dart/conduit/pull/70), please check it out when time allows.

**Comment by j4qfrost** [comment]
Closed with #153
